ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/SearchablePluginsManager.java

```diff
@@ -400,37 +406,52 @@ public void removePluginModelListener(IPluginModelListener listener) {
 	}
 
 	private void saveStates() {
-		// persist state
-		IWorkspaceRoot root = PDECore.getWorkspace().getRoot();
-		IProject project = root.getProject(PROXY_PROJECT_NAME);
-		if (project.exists() && project.isOpen()) {
-			// modify the .searchable file only if there is any change
-			Set<String> loadedStates = loadStates();
-			String propertyToSave;
-			synchronized (fPluginIdSet) {
-				if (loadedStates.equals(fPluginIdSet)) {
-					return;
+		WorkspaceJob job = new WorkspaceJob(PDECoreMessages.SearchablePluginsManager_SaveStates) { 
+			@Override
+			public IStatus runInWorkspace(IProgressMonitor monitor) {
+			// persist state
+			IWorkspaceRoot root = PDECore.getWorkspace().getRoot();
+			IProject project = root.getProject(PROXY_PROJECT_NAME);
+			if (project.exists() && project.isOpen()) {
+				// modify the .searchable file only if there is any change
+				Set<String> loadedStates = loadStates();
+				String propertyToSave;
+				synchronized (fPluginIdSet) {
+					if (loadedStates.equals(fPluginIdSet)) {
+							return Status.OK_STATUS;
+					}
+					propertyToSave = fPluginIdSet.stream().collect(Collectors.joining(",")); //$NON-NLS-1$
 				}
-				propertyToSave = fPluginIdSet.stream().collect(Collectors.joining(",")); //$NON-NLS-1$
-			}
-			IFile file = project.getFile(PROXY_FILE_NAME);
-			Properties properties = new Properties();
-			properties.setProperty(KEY, propertyToSave);
-			try (ByteArrayOutputStream outStream = new ByteArrayOutputStream()) {
-				properties.store(outStream, ""); //$NON-NLS-1$
-				outStream.flush();
-				outStream.close();
-				try (ByteArrayInputStream inStream = new ByteArrayInputStream(outStream.toByteArray())) {
-					if (file.exists()) {
-						file.setContents(inStream, true, false, new NullProgressMonitor());
-					} else {
-						file.create(inStream, true, new NullProgressMonitor());
+				IFile file = project.getFile(PROXY_FILE_NAME);
+				Properties properties = new Properties();
+				properties.setProperty(KEY, propertyToSave);
+				try (ByteArrayOutputStream outStream = new ByteArrayOutputStream()) {
+					properties.store(outStream, ""); //$NON-NLS-1$
+					outStream.flush();
+					outStream.close();
+					try (ByteArrayInputStream inStream = new ByteArrayInputStream(outStream.toByteArray())) {
+						if (file.exists()) {
+							file.setContents(inStream, true, false, new NullProgressMonitor());
+						} else {
+							file.create(inStream, true, new NullProgressMonitor());
+						}
+							return Status.OK_STATUS;
 					}
+				} catch (IOException | CoreException e) {
+					PDECore.log(e);
 				}
-			} catch (IOException | CoreException e) {
-				PDECore.log(e);
 			}
-		}
+				return Status.OK_STATUS;
+			}
+		};
+			job.setSystem(true);
+			job.setRule(ResourcesPlugin.getWorkspace().getRoot());
+			Display display = Display.getCurrent();
+			if (display != null) {
+				PlatformUI.getWorkbench().getProgressService().showInDialog(display.getActiveShell(),
```

Comment:
Please don't use that API (anymore). We won't too many explicit popups, especially here for that operation